src/viur/core/modules/file.py

```diff
@@ -449,6 +493,53 @@ def hmac_sign(data: t.Any) -> str:
     def hmac_verify(data: t.Any, signature: str) -> bool:
         return hmac.compare_digest(File.hmac_sign(data.encode("ASCII")), signature)
 
+    @staticmethod
+    def create_serve_parameters(serving_url: str) -> t.Iterable[str] | None:
+        """
+        Splits a serving URL into its components, used by serve function.
+        :param serving_url: the serving URL to be split
+        :return: return a tuple of host and key
+        """
+        regex_result = re.match(
+            r"^https:\/\/(.*?)\.googleusercontent\.com\/(.*?)$",
+            serving_url
+        )
+
+        if regex_result:
+            return regex_result.groups()
+        return regex_result
+
+    @staticmethod
+    def create_serve_url(
+        serving_url: str,
+        size: t.Optional[int] = None,
+        filename: t.Optional[str] = None,
+        options: str = "",
+        download: bool = False
+    ) -> str:
+
+        serve_url = f"{File.SERVE_URL_PREFIX}"
+
+        serving_params = File.create_serve_parameters(serving_url)
+        if not serving_params:
+            raise errors.UnprocessableEntity(f"Invalid serving_url {serving_url!r} provided")
+
+        serve_url += f"/{'/'.join(serving_params)}"
+
+        path_parameters = [f"{x}" for x in [size, filename] if x is not None]
```

Comment:
This does not work, if `filename` is provided, but not `size`. Then, `filename` would be used as `size`.

I would suggest to switch the order of the `filename` and `size` parameter in `serve()` and here in `create_serve_url()` and append `size` always as keyword argument in the query string. 

To have the filename as last positional has btw. the advantage, that browsers use the last url path segment if a file is displayed inline for the save dialog and tab title.

Reply:
You're right, need to fix this.

Reply:
Fixed by a94b9226